src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -212,49 +339,46 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
 	region.bufferOffset = 0u;
 	region.bufferImageHeight = 0u;
 	region.imageOffset = { 0u, 0u, 0u };
-	region.imageExtent = image->getCreationParameters().extent;
+	region.imageExtent = imgInfo.extent;
 
 	// read image
 	size_t endBufferSize = {};
 	
 	switch (header.ImageType)
 	{
-		case 1: // Uncompressed color-mapped image
-		case 2: // Uncompressed RGB image
-		case 3: // Uncompressed grayscale image
-			{
-				region.bufferRowLength = calcPitchInBlocks(region.imageExtent.width, getTexelOrBlockBytesize(EF_R8G8B8_SRGB));
-				const int32_t imageSize = endBufferSize = region.imageExtent.height * region.bufferRowLength * bytesPerTexel;
-				texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(imageSize);
-				_file->read(texelBuffer->getPointer(), imageSize);
-			}
-			break;
-		
-		case 10: // Run-length encoded (RLE) true color image
+		case STIT_NONE:
+		{
+			os::Printer::log("The given TGA doesn't have image data", _file->getFileName().c_str(), ELL_ERROR);
+			return {};
+		}
+		case STIT_UNCOMPRESSED_COLOR_MAPPED_IMAGE: _IRR_FALLTHROUGH;
```

Comment:
color mapped is an error, since you're not using `colorMap` yet

Reply:
it is as it was

Reply:
deleted in upcoming commit